crates/runtime/src/tools/memory/mod.rs

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2024 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use std::sync::Arc;
+
+use arrow::array::{ArrayRef, RecordBatch, StringArray, TimestampSecondArray};
+use arrow_schema::{ArrowError, DataType, Field, Schema, TimeUnit};
+use uuid::Uuid;
+
+pub mod catalog;
+pub mod load;
+pub mod store;
+
+pub static DEFAULT_MEMORY_TABLE: &str = "spice.public.store";
+
+pub struct MemoryTableElement {
+    pub id: Uuid,
+    pub value: String,
+    pub created_by: Option<String>,
+    pub created_at: i64, // Unix timestamp in Seconds
+}
+
+pub fn try_from(data: &[MemoryTableElement]) -> Result<RecordBatch, ArrowError> {
+    let ids = StringArray::from_iter_values(data.iter().map(|d| d.id.to_string()));
+    let values = StringArray::from_iter_values(data.iter().map(|d| d.value.as_str()));
+    let created_by = data
+        .iter()
+        .map(|d| d.created_by.as_deref())
+        .collect::<StringArray>();
+    let created_at: TimestampSecondArray =
+        TimestampSecondArray::from(data.iter().map(|e| e.created_at).collect::<Vec<_>>());
+
+    let schema = Arc::new(Schema::new(vec![
```

Comment:
This schema should come from the same place the data connector gets it, to ensure they stay in sync